### ISSUE TITLE
predict_linear(<metric>, ?)

### ISSUE THREAD

**Issue by Licenser** [dalmatinerdb/dalmatinerdb#36]

**Comment by sacreman** [comment]
A prediction X number of seconds into the future for a series that uses simple linear regression.

In reality this is probably only going to be useful for disk space. Maybe memory if you are using Java :)

**Comment by Licenser** [comment]
how would that work in a query you query between T0 and T1, and then extend the end with T2?

Or you queyr between T1 and T2 and 'predict' values with little confidence with a linear algorithm? 

My gut feeling tells me option 2 is more universal, so it might be something like

fill_linear(<metric>, <minimal confidence>)? 

DDB will fill up with 'empties' if you querry beyond the last known point.

**Comment by sacreman** [comment]
Yes, I think it would probably be a case of returning the query data between two points and then merging a second query that linearly predicts and fills the gaps into the future based on the first.

Prometheus has this function https://prometheus.io/docs/querying/functions/#predict_linear()
